x-pack/plugins/security_solution/public/cloud_security_posture/components/csp_details/vulnerabilities_findings_details_table.tsx

```diff
@@ -0,0 +1,217 @@
+  CspVulnerabilityFinding,
+  'vulnerability' | 'resource'
+>;
+
+interface VulnerabilitiesPackage extends Vulnerability {
+  package: {
+    name: string;
+  };
+}
+
+/**
+ * Insights view displayed in the document details expandable flyout left section
+ */
+export const VulnerabilitiesFindingsDetailsTable = memo(({ queryName }: { queryName: string }) => {
+  //
+  const { data } = useVulnerabilitiesFindings({
+    query: buildEntityFlyoutPreviewQuery('host.name', queryName),
+    sort: [],
+    enabled: true,
+    pageSize: 1,
+  });
+
+  const { CRITICAL = 0, HIGH = 0, MEDIUM = 0, LOW = 0, UNKNOWN = 0 } = data?.count || {};
+
+  const [pageIndex, setPageIndex] = useState(0);
+  const [pageSize, setPageSize] = useState(10);
+
+  const findingsPagination = (findings: VulnerabilitiesFindingDetailFields[]) => {
+    let pageOfItems;
+
+    if (!pageIndex && !pageSize) {
+      pageOfItems = findings;
+    } else {
+      const startIndex = pageIndex * pageSize;
+      pageOfItems = findings?.slice(startIndex, Math.min(startIndex + pageSize, findings?.length));
+    }
+
+    return {
+      pageOfItems,
+      totalItemCount: findings?.length,
+    };
+  };
+
+  const { pageOfItems, totalItemCount } = findingsPagination(data?.rows || []);
+
+  const pagination = {
+    pageIndex,
+    pageSize,
+    totalItemCount,
+    pageSizeOptions: [10, 25, 100],
+  };
+
+  const onTableChange = ({ page }: Criteria<VulnerabilitiesFindingDetailFields>) => {
+    if (page) {
+      const { index, size } = page;
+      setPageIndex(index);
+      setPageSize(size);
+    }
+  };
+
+  const navToVulnerabilities = useNavigateVulnerabilities();
+
+  const navToVulnerabilitiesByName = (name: string, queryField: 'host.name' | 'user.name') => {
+    navToVulnerabilities({ [queryField]: name });
+  };
+
+  const navToVulnerabilityByVulnerabilityAndResourceId = (
+    vulnerabilityId: string,
+    resourceId: string
+  ) => {
+    navToVulnerabilities({
+      'vulnerability.id': vulnerabilityId,
+      'resource.id': resourceId,
+    });
+  };
+
+  const columns: Array<EuiBasicTableColumn<VulnerabilitiesFindingDetailFields>> = [
+    {
+      field: 'vulnerability',
+      name: '',
+      width: '5%',
+      render: (
+        vulnerability: VulnerabilitiesPackage,
+        finding: VulnerabilitiesFindingDetailFields
+      ) => (
+        <EuiLink
+          onClick={() => {
+            navToVulnerabilityByVulnerabilityAndResourceId(
+              vulnerability?.id,
+              finding?.resource?.id || ''
+            );
+          }}
+        >
+          <EuiIcon type={'popout'} />
+        </EuiLink>
+      ),
+    },
+    {
+      field: 'vulnerability',
+      render: (vulnerability: Vulnerability) => <EuiText size="s">{vulnerability?.id}</EuiText>,
+      name: i18n.translate(
+        'xpack.securitySolution.flyout.left.insights.vulnerability.table.resultColumnName',
+        {
+          defaultMessage: 'Vulnerability',
+        }
```

Comment:
another general comment that helps with readability, the entire pr can one line the default messages like so:

```suggestion
        { defaultMessage: 'Vulnerability' }
```

just inline one of the brackets and lint will do the rest